client/extensions/woocommerce/app/order/order-details.js

```diff
@@ -1,36 +1,87 @@
 /**
  * External dependencies
  */
+import { find } from 'lodash';
+import Gridicon from 'gridicons';
 import { localize } from 'i18n-calypso';
 import React, { Component, PropTypes } from 'react';
-import Gridicon from 'gridicons';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
 import Card from 'components/card';
+import FormSelect from 'components/forms/form-select';
 import OrderDetailsTable from './order-details-table';
 import OrderRefundCard from './order-refund-card';
 import SectionHeader from 'components/section-header';
 
 class OrderDetails extends Component {
 	static propTypes = {
+		onUpdate: PropTypes.func,
 		order: PropTypes.object.isRequired,
 		site: PropTypes.shape( {
 			ID: PropTypes.number.isRequired,
 			slug: PropTypes.string.isRequired,
 		} ),
 	}
 
-	state = {
-		showRefundDialog: false,
+	constructor( props ) {
+		super( props );
+		this.state = {
+			status: this.props.order.status || false,
+		};
 	}
 
-	toggleRefundDialog = () => {
-		this.setState( {
-			showRefundDialog: ! this.state.showRefundDialog,
-		} );
+	updateStatus = ( event ) => {
+		this.setState( { status: event.target.value } );
+		// Send the order back to the parent component
+		this.props.onUpdate( { status: event.target.value } );
+	}
+
+	renderStatus = () => {
+		const { order, translate } = this.props;
+		const classes = `order__status is-${ order.status }`;
+		// TODO: create a helper function for status labels
+		const statuses = [ {
+			value: 'pending',
+			name: translate( 'Pending payment' ),
+		}, {
+			value: 'processing',
+			name: translate( 'Processing' ),
+		}, {
+			value: 'on-hold',
+			name: translate( 'On Hold' ),
+		}, {
+			value: 'completed',
+			name: translate( 'Completed' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **52** times:
`translate( '&laquo; completed' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/woo-orders-action-header)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).